src/GraphQL/Utility/FileUpload.php

```diff
@@ -0,0 +1,254 @@
+   *
+   * @param \Symfony\Component\HttpFoundation\File\UploadedFile $file
+   *   The file entity to upload.
+   * @param array $settings
+   *   Fle settings as specified in regular file field config.
+   *
+   * @return \Drupal\file\FileInterface|bool
+   *   The file entity or false in case of unsuccessful upload.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function createTemporaryFileUpload(UploadedFile $file, array $settings) {
+    $this->violations = [];
+    // Check for file upload errors and return FALSE for this file if a lower
+    // level system error occurred.
+    // @see http://php.net/manual/features.file-upload.errors.php.
+    switch ($file->getError()) {
+      case UPLOAD_ERR_INI_SIZE:
+      case UPLOAD_ERR_FORM_SIZE:
+        $max_upload_size = format_size($this->getMaxUploadSize($settings));
+        $this->registerViolations($this->t('The file %file could not be saved because it exceeds %maxsize, the maximum allowed size for uploads.', ['%file' => $file->getFilename(), '%maxsize' => $max_upload_size]));
+        $this->logger->error(sprintf('The file "%s" could not be saved because it exceeds "%s", the maximum allowed size for uploads.', $file->getFilename(), $max_upload_size));
+        return FALSE;
+
+      case UPLOAD_ERR_PARTIAL:
+      case UPLOAD_ERR_NO_FILE:
+        $this->registerViolations($this->t('The file "@file" could not be saved because the upload did not complete.', ['@file' => $file->getFilename()]));
+        $this->logger->error(sprintf('The file "%s" could not be saved because the upload did not complete.', $file->getFilename()));
+        return FALSE;
+
+      case UPLOAD_ERR_OK:
+        // Final check that this is a valid upload, if it isn't, use the
+        // default error handler.
+        if (is_uploaded_file($file->getRealPath())) {
+          break;
+        }
+
+      default:
+        $this->registerViolations($this->t('Unknown error while uploading the file "@file".', ['@file' => $file->getFilename()]));
+        $this->logger->error(sprintf('Unknown error while uploading the file "%s".', $file->getFilename()));
+        return FALSE;
+    }
+    $upload_dir = $settings['uri_scheme'] . '://' . trim($settings['file_directory'], '/');
+    if (!$this->fileSystem->prepareDirectory($upload_dir, FileSystem::CREATE_DIRECTORY)) {
+      $this->registerViolations($this->t('Unknown error while uploading the file "@file".', ['@file' => $file->getFilename()]));
+      $this->logger->error(sprintf('Could not create directory "%s".', $upload_dir));
+      return FALSE;
+    }
+    $name = $file->getClientOriginalName();
+    $mime = $this->mimeTypeGuesser->guess($name);
+    $destination = $this->fileSystem->getDestinationFilename("{$upload_dir}/{$name}", $this->fileSystem::EXISTS_RENAME);
+    // Begin building file entity.
```

Comment:
Can you add some whitespace around the code in this method here? It's already a long method body. Would help readability!